### DISCUSSION TITLE
Feature Request: Export Calendar Event Link Generating Functions in atcb-links

### DISCUSSION THREAD

**Discussion by williscool** [add2cal/add-to-calendar-button#461] [Ideas / Requests]
Title pretty much says it all but I think it would be very useful to have. 
https://github.com/add2cal/add-to-calendar-button/blob/main/src/atcb-links.js

As a separate library when you want to build a custom UI around building calendar events but want to leverage all the great work that this library has done learning how to build the links properly. Preferably with typescript type definitions 🥺.

Similar to

https://github.com/add2cal/timezones-ical-library

**Comment by williscool** [comment]
As you can see 

https://npmtrends.com/add-to-calendar-button-react-vs-add-to-calendar-button-vs-calendar-link-vs-generate-calendar-url-vs-react-add-to-calendar-vs-react-add-to-calendar-hoc

https://github.com/AnandChowdhary/calendar-link

Has taken off recently doing just that. But it's not quite as feature rich as atcb's link generators.

**Comment by jekuer** [comment]
Hi @williscool ,

Thank you for your comment!
I updated your link, since it was referring to the wrong package.

Regarding the feature request:
The current approach does not really support this as this is no library, but rather some easy to use e2e widget/tool/snippet/component.
Another very big issues is that there is no stable scheme for the url. It highly depends on the user's setup and in some cases can only be resolved together with smart UI. For example, multi-date events are only working for iCal. For Google we show a list of the individual events instead. This cannot be mitigated by simply generating links.
Long story short: Such thing as a "link generator" is a myth - at least when you want to support more functionality than the mentioned package.
Therefore, the better approach would be to customize the given UI (via the available options, see https://add-to-calendar-button.com/advanced-use#case-1 for details). This is the only good approach atm.

We are currently working on a managed service - eventually, there are new possibilities coming up with this.
Additionally, if everything goes as planned, there will be a bigger refactoring next year.
I am moving this to the discussions area, so we do not forget about it and can consider it when reworking everything.

All the best,
Jens
